src/main/kotlin/fi/metatavu/muisti/api/PageLayoutsApiImpl.kt

```diff
@@ -47,9 +50,20 @@ class PageLayoutsApiImpl: PageLayoutsApi, AbstractApi() {
         return createOk(pageLayoutTranslator.translate(pageLayout))
     }
 
-    override fun listPageLayouts(): Response {
-        val pageLayouts = pageLayoutController.listPageLayouts()
-        return createOk(pageLayouts.map (pageLayoutTranslator::translate))
+    override fun listPageLayouts(deviceModelId: UUID?, screenOrientation: String?): Response? {
+        val result = if (deviceModelId !== null && screenOrientation !== null) {
```

Comment:
This code looks a bit messy and contains logic that should be moved into the controller